cmd/cloud-controller-manager/nodeipamcontroller.go

```diff
@@ -146,10 +149,21 @@ func startNodeIpamController(ccmConfig *cloudcontrollerconfig.CompletedConfig, n
 	// get list of node cidr mask sizes
 	nodeCIDRMaskSizes := getNodeCIDRMaskSizes(clusterCIDRs, nodeCIDRMaskSizeIPv4, nodeCIDRMaskSizeIPv6)
 
+	kubeConfig := ccmConfig.Complete().Kubeconfig
+	kubeConfig.ContentType = "application/json" // required to serialize Networks to json
+	networkClient, err := networkclientset.NewForConfig(kubeConfig)
+	if err != nil {
+		return nil, false, err
+	}
+	nwInfFactory := networkinformers.NewSharedInformerFactory(networkClient, 30*time.Second)
```

Comment:
What would happen if the network CRD and GKENetworkParamsCRD was not loaded? would this succeed or throw an error? We need to be careful not to break non multi networking clusters.

Reply:
I've tested this. It does not affect any controller behaviour, podCIDRs are allotted just fine. As @aojea mentioned, reflector will fail to list the objects with the following error:

```
Listing and watching *v1alpha1.GKENetworkParamSet from vendor/k8s.io/cloud-provider-gcp/crd/client/network/informers/externalversions/factory.go:117
W1215 18:43:46.012382       1 reflector.go:424] vendor/k8s.io/cloud-provider-gcp/crd/client/network/informers/externalversions/factory.go:117: failed to list *v1alpha1.GKENetworkParamSet: the server could not find the requested resource (get gkenetworkparamsets.networking.gke.io)
E1215 18:43:46.012693       1 reflector.go:140] vendor/k8s.io/cloud-provider-gcp/crd/client/network/informers/externalversions/factory.go:117: Failed to watch *v1alpha1.GKENetworkParamSet: failed to list *v1alpha1.GKENetworkParamSet: the server could not find the requested resource (get gkenetworkparamsets.networking.gke.io)

Listing and watching *v1.Network from vendor/k8s.io/cloud-provider-gcp/crd/client/network/informers/externalversions/factory.go:117
W1215 18:44:01.401237       1 reflector.go:424] vendor/k8s.io/cloud-provider-gcp/crd/client/network/informers/externalversions/factory.go:117: failed to list *v1.Network: the server could not find the requested resource (get networks.networking.gke.io)
E1215 18:44:01.401277       1 reflector.go:140] vendor/k8s.io/cloud-provider-gcp/crd/client/network/informers/externalversions/factory.go:117: Failed to watch *v1.Network: failed to list *v1.Network: the server could not find the requested resource (get networks.networking.gke.io)
```

These logs will remain in the CCM for as long as the cluster is running.

Reply:
Will it flood the controller logs with these errors? Would be great to avoid erroring out in irrelevant case.